### DISCUSSION TITLE
How to disable dragging?

### DISCUSSION THREAD

**Discussion by DavraYoung** [prevwong/craft.js#244] [Q&A]
We are using `0.2.0-alpha.23` craftjs version. We try to implement free positioning of craftjs node elements based on the properties. So depending on component props it will use either  CraftJS Drag or React Draggale drag. And now we faced a problem: 
**We cannot disable drag handler on our component, after it was initialized as `ref={ref=>connect(drag(ref))}`.** 

We tried changing react key, so that ref function gets called when props is changed, so that we can conditionally run `connect(drag(ref))` or just `connect(ref)`, but dragging is still handled on CraftJS side. How can we dynamically disable Craft JS dragging? 

Using craft.rules.canDrag function does not help, as it only prevents component from being dropped (for some reason), whereas in our case we want to completely disable drag control, so that React Draggable can handle it appropriately.

**Comment by DavraYoung** [comment]
It's turns out, if `canDrag` function returns once true, then the behaviour of dragging is changed, and canDrag acts like canDrop. How can we avoid this?

**Comment by MarcDAFrame** [comment]
any update on this?

**Comment by luc-hn** [comment]
any update on this?
